### ISSUE TITLE
SchemaException when type has Equals method

### ISSUE THREAD

**Issue by Dolfik1** [ChilliCream/hotchocolate#229]
I implemented method `Equals` in my class:
```csharp
public class Test
{
    public override Equals(object obj) => true;
}
```

Then I created ObjectType:
```csharp
public class TestType
  : ObjectType<Test>
{
  public override void Configure(IObjectTypeDescriptor<Test> desc)
  {
    desc.Field(x => x.Equals(default)).Ignore();
  }
}
```

When I run project I'm getting following error:
```HotChocolate.SchemaException : The type `System.Object` of field `Test.obj` could not be resolved to a valid schema type. - Type: Test
   at HotChocolate.Schema.CreateSchema(SchemaContext context, Action`1 configure)
   at Renta.Tests.Mock.SchemaMock.CreateSchema(IServiceProvider provider) in ```

This is a bug?

**Comment by michaelstaib** [comment]
I will check this. In your case you want to ignore equals and it ends up in a schema exception. I would say it looks like a bug. Let me get back to you on that.

**Comment by michaelstaib** [comment]
It is a bug ... I will fix it and make a release. I think you will have the fix by tomorrow.

**Comment by michaelstaib** [comment]
Fixed with #230

**Comment by michaelstaib** [comment]
@Dolfik1: So, there was an issue with the reflection utilities when the expression of an field was extracted the extracted member always referred to the member in the base class. So, for the implicit field discoverer was not clear that the member was already handled. 

I have fixed this now and will include it in version 0.4.4. I started the build process, should be ready in the next 30 minutes.

The issue actually would have made problems with all kinds of overridden members:

```csharp

public class TestBase
{
    public virtual string A { get; set; }
}

public class Test 
    : TestBase
{
    public override string A { get; set; }
}

public class TestType
  : ObjectType<Test>
{
  public override void Configure(IObjectTypeDescriptor<Test> desc)
  {
    desc.Field(x => x.A).Ignore();
  }
}

```

In the above example the ignore wouldn't have been handled.

Thanks, for your help, really appreciate your feedback 👍
